presto-cli/src/main/java/com/facebook/presto/cli/ErrorMessages.java

```diff
@@ -23,22 +23,75 @@
 import com.google.common.collect.ImmutableList;
 import org.fusesource.jansi.Ansi;
 
+import java.io.EOFException;
 import java.io.PrintWriter;
 import java.io.StringWriter;
+import java.net.ConnectException;
 import java.util.List;
 
 import static com.facebook.presto.cli.ConsolePrinter.REAL_TERMINAL;
 import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Throwables.getCausalChain;
 import static java.lang.String.format;
 
 public class ErrorMessages
 {
+    private static final String PRESTO_COORDINATOR_NOT_FOUND = "There was a problem with a response from Presto Coordinator.\n";
+
     private static final String TECHNICAL_DETAILS_HEADER = "\n=========   TECHNICAL DETAILS   =========\n";
     private static final String SESSION_INTRO = "[ Session information ]\n";
     private static final String ERROR_MESSAGE_INTRO = "[ Error message ]\n";
     private static final String STACKTRACE_INTRO = "[ Stack trace ]\n";
     private static final String TECHNICAL_DETAILS_END = "========= TECHNICAL DETAILS END =========\n\n";
 
+    private enum Tip {
+        VERIFY_PRESTO_RUNNING("Verify that Presto is running on %s."),
+        DEFINE_SERVER_AS_CLI_PARAM("Use '--server' argument when starting Presto CLI to define server host and port."),
```

Comment:
One concern about these is that, at FB, we wrap the CLI with a launcher that sets some parameter (server, etc. after looking up the corresponding cluster given a symbolic name). These hints are not relevant to our users, and may actually be confusing. 

E.g.,
- they have no way to verify presto is running on xxx. That's something managed by the Presto team. The message we would show in this case would be "Check XXX internal group for announcements about maintenance", or "check XXX status page for availability information".
- they can't provide the --server argument. That's handled by the wrapper. Setting it would cause the wrapper to fail to run.
- they can't check network conditions between client and server. That's infrastructure they have no control over or knowledge of how it works.
- it's unlikely they'll understand the output of --debug or be able to make any sense of it.